reana_client/cli/workflow.py

```diff
@@ -1458,3 +1464,84 @@ def workflow_close_interactive_session(workflow, access_token):  # noqa: D301
             sys.exit(1)
     else:
         display_message("Cannot find workflow {} ".format(workflow), msg_type="error")
+
+
+@workflow_sharing_group.command("share-add")
+@check_connection
+@add_workflow_option
+@add_access_token_options
+@click.option(
+    "-u",
+    "--user",
+    "users",
+    multiple=True,
+    help="Users to share the workflow with.",
+    required=True,
+)
+@click.option(
+    "-m",
+    "--message",
+    help="Optional message that is sent to the user(s) with the sharing invitation.",
+)
+@click.option(
+    "-v",
+    "--valid-until",
+    help="Optional date when access to the workflow will expire for the given user(s) (format: YYYY-MM-DD).",
+)
+@click.pass_context
+def share_workflow_add(
+    ctx, workflow, access_token, users, message, valid_until
+):  # noqa D412
+    """Share a workflow with other users (read-only).
+
+    The `share-add` command allows sharing a workflow with other users. The
+    users will be able to view the workflow but not modify it.
+
+    Examples:
+
+    $ reana-client share-add -w myanalysis.42 --user bob@example.org
+
+    $ reana-client share-add -w myanalysis.42 --user bob@example.org --user cecile@example.org --message "Please review my analysis" --valid-until 2025-12-31
+    """
+    from reana_client.api.client import share_workflow
+
+    share_errors = []
+    shared_users = []
+
+    if workflow:
+        try:
+            for user in users:
+                try:
+                    logging.info(f"Sharing workflow {workflow} with user {user}")
+                    share_workflow(
+                        workflow,
+                        user,
+                        access_token,
+                        message=message,
+                        valid_until=valid_until,
+                    )
+                    shared_users.append(user)
+                except Exception as e:
+                    share_errors.append(
+                        f"Failed to share {workflow} with {user}: {str(e)}"
+                    )
+                    logging.debug(traceback.format_exc())
+        except Exception as e:
+            logging.debug(traceback.format_exc())
+            logging.debug(str(e))
+            display_message(
+                "An error occurred while sharing workflow:\n{}".format(str(e)),
+                msg_type="error",
+            )
+
+        if shared_users:
+            display_message(
+                f"{workflow} is now read-only shared with {', '.join(shared_users)}",
+                msg_type="success",
+            )
+        if share_errors:
+            for error in share_errors:
+                display_message(error, msg_type="error")
```

Comment:
If there are errors (here or before), then we reana-client should exit with non-zero exit code

Reply:
This would cancel out any subsequent errors. E.g. imagine someone shares with 20 people at a time where some emails have typos; the current implementation would then show all errors after 1 command execution, while the implementation you suggest requires the user to run the command again after each error to find the next error. 

We can still change this if you prefer the latter scenario for a reason that I'm not seeing.

Reply:
Yeah, it's good that the errors are all shown at the end! What I meant was to add something like
```python
if share_errors:
    sys.exit(1)
```
at the end of the command, so that in case of errors reana-client returns an appropriate status code. No need to call sys.exit it when the error happens.

Reply:
Agreed :+1: